src/coreclr/jit/llvm.cpp

```diff
@@ -841,26 +852,131 @@ llvm::Value* Llvm::buildUserFuncCall(GenTreeCall* call)
     return mapGenTreeToValue(call, llvmCall);
 }
 
-void Llvm::buildCall(GenTree* node)
+llvm::FunctionType* Llvm::buildHelperLlvmFunctionType(GenTreeCall* call, bool withShadowStack)
 {
-    GenTreeCall* call = node->AsCall();
-    if (call->gtCallType == CT_HELPER)
+    Type* retLlvmType = getLlvmTypeForVarType(call->TypeGet());
+    std::vector<llvm::Type*> argVec;
+
+    if (withShadowStack)
```

Comment:
We insert the shadow stack in IR for user call, seems like we should do that for the helper calls as well?

Reply:
Its been talked about, but currently we only add the return slot to the IR for user calls.  The shadow stack arg itself has not been done yet.  Its added for user calls here https://github.com/dotnet/runtimelab/blob/d623b2be57be73364ecf853a0fa31e7e9582b4ee/src/coreclr/jit/llvm.cpp#L797-L801  Agree though that it would be good to do.  I've got a couple of PRs queued up behind this one, then I can look at moving more things to lowering, if that's ok?

Reply:
> then I can look at moving more things to lowering, if that's ok?

Of course.